src/commands/paranet/paranet-sync-command.js

```diff
@@ -464,55 +470,40 @@ class ParanetSyncCommand extends Command {
                     continue;
                 }
 
-                filteredKAs.push([
+                pendingFilteredKAs.push([
                     ual,
                     blockchain,
                     knowledgeAssetStorageContract,
                     knowledgeAssetTokenId,
                 ]);
-            }
-
-            kasToSync.push(...filteredKAs);
-        }
-
-        const results = [];
-
-        // Loop through kasToSync in batches
-        for (let i = 0; i < kasToSync.length; i += PARANET_SYNC_KA_COUNT) {
-            // Get the current batch
-            const batch = kasToSync.slice(i, i + PARANET_SYNC_KA_COUNT);
-
-            // Map the batch to an array of promises
-            const promises = batch.map(
-                ([
-                    ual,
-                    knowledgeAssetBlockchain,
-                    knowledgeAssetStorageContract,
-                    knowledgeAssetTokenId,
-                ]) =>
-                    this.syncAsset(
-                        ual,
-                        knowledgeAssetBlockchain,
-                        knowledgeAssetStorageContract,
-                        knowledgeAssetTokenId,
-                        paranetUAL,
-                        paranetId,
-                        paranetMetadata,
-                        paranetNodesAccessPolicy,
-                        operationId,
-                        false, // removeMissingAssetRecord
-                    ),
-            );
 
-            // Await the promises in the current batch
-            const batchResults = await Promise.all(promises);
+                // Process batches when we have enough KAs
+                while (pendingFilteredKAs.length >= PARANET_SYNC_KA_COUNT) {
+                    const kasToSync = pendingFilteredKAs.splice(0, PARANET_SYNC_KA_COUNT);
```

Comment:
KAs to be synced are never removed from ```pendingFilteredKAs``` we if it stays like this it will only take first ```PARANET_SYNC_KA_COUNT``` asset from start of the array